source/agora/node/Ledger.d

```diff
@@ -1251,15 +1262,24 @@ version (unittest)
 
             auto stateDB = new ManagedDatabase(":memory:");
             auto cacheDB = new ManagedDatabase(":memory:");
+            EnrollmentManager enroll_man = new EnrollmentManager(stateDB, cacheDB, key_pair, params);
             super(params,
                 new Engine(TestStackMaxTotalSize, TestStackMaxItemSize),
                 new UTXOSet(stateDB),
-                new MemBlockStorage(blocks),
-                new EnrollmentManager(stateDB, cacheDB, key_pair, params),
+                new MemBlockStorage(blocks ? blocks.takeExactly(1) : null),
+                enroll_man,
                 new TransactionPool(cacheDB),
                 new FeeManager(stateDB, params),
                 mock_clock,
                 block_time_offset_tolerance_dur, null);
+            if (blocks)
+            {
+                blocks[1 .. $].map!(b => b.header.height).each!(h => simulatePreimages(this.log, enroll_man, genesis_validator_keys, h));
+                blocks[1 .. $].each!(b => {
+                    if (this.validateBlock(b) == null)  // i.e. no invalid reason
+                        this.addValidatedBlock(b);
+                }());
+            }
```

Comment:
Same thing here, feed the Ledger valid block instead of patching them inside the ctor

Reply:
Updated.